homeassistant/components/tod/binary_sensor.py

```diff
@@ -100,40 +104,53 @@ class TodSensor(BinarySensorEntity):
 
     _attr_should_poll = False
 
-    def __init__(self, name, after, after_offset, before, before_offset, unique_id):
+    def __init__(
+        self,
+        name: str,
+        after: time,
+        after_offset: timedelta,
+        before: time,
+        before_offset: timedelta,
+        unique_id: str | None,
+    ) -> None:
         """Init the ToD Sensor..."""
         self._attr_unique_id = unique_id
-        self._name = name
-        self._time_before = self._time_after = self._next_update = None
+        self._attr_name = name
+        self._time_before: datetime | None = None
+        self._time_after: datetime | None = None
+        self._next_update: datetime | None = None
         self._after_offset = after_offset
         self._before_offset = before_offset
         self._before = before
         self._after = after
-        self._unsub_update: Callable[[], None] = None
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+        self._unsub_update: Callable[[], None] | None = None
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return True is sensor is on."""
+        if TYPE_CHECKING:
+            assert self._time_after is not None
+            assert self._time_before is not None
         if self._time_after < self._time_before:
             return self._time_after <= dt_util.utcnow() < self._time_before
         return False
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return the state attributes of the sensor."""
-        time_zone = dt_util.get_time_zone(self.hass.config.time_zone)
-        return {
-            ATTR_AFTER: self._time_after.astimezone(time_zone).isoformat(),
-            ATTR_BEFORE: self._time_before.astimezone(time_zone).isoformat(),
-            ATTR_NEXT_UPDATE: self._next_update.astimezone(time_zone).isoformat(),
-        }
-
-    def _naive_time_to_utc_datetime(self, naive_time):
+        if TYPE_CHECKING:
+            assert self._time_after is not None
+            assert self._time_before is not None
+            assert self._next_update is not None
+        if time_zone := dt_util.get_time_zone(self.hass.config.time_zone):
```

Comment:
Side note: If more than one line gets indented after a condition, try inverting the condition and see if less lines are indented. Guard clause is a nice tool to make code easier to read.